python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -198,20 +210,51 @@ def getWNScriptDownloadCommand(self, indent):
 
         return script
 
+    def _mkdir(self, massStoragePath):
+        """
+        Creates a folder on the mass Storage corresponding to the given path
+        Args:
+            massStoragePath (str): This is the path we want to make if it doesn't exist.
+        """
+
+        massStorageConfig = getConfig('Output')['MassStorageFile']['uploadOptions']
+        mkdir_cmd = massStorageConfig['mkdir_cmd']
+        ls_cmd = massStorageConfig['ls_cmd']
+
+        # create the last directory (if not exist) from the config path
+        pathToDirName = os.path.dirname(massStoragePath)
+        dirName = os.path.basename(massStoragePath)
+
+        directoryExists = False
+
+        (exitcode, mystdout, mystderr) = self.execSyscmdSubprocess('%s %s' % (ls_cmd, escapeWhiteSpace(pathToDirName)))
+        if exitcode != 0:
+            #self.handleUploadFailure(mystderr, '1) %s %s' % (ls_cmd, pathToDirName))
+            #raise GangaException(mystderr)
+            directoryExists = False
+
+        for directory in mystdout.split('\n'):
+            if directory.strip() == dirName:
+                directoryExists = True
+                break
+
+        if not directoryExists:
+            (exitcode, mystdout, mystderr) = self.execSyscmdSubprocess('%s -p %s' % (mkdir_cmd, escapeWhiteSpace(massStoragePath)))
+            if exitcode != 0:
+                self.handleUploadFailure(mystderr, '2) %s %s' % (mkdir_cmd, massStoragePath))
+                raise GangaException(mystderr)
```

Comment:
If this is a `GangaException` intended for end-users then it would be helpful to give a user-facing message in addition to the raw output from the mkdir command.